### PR TITLE
Feature/std 62/3 d map

### PR DESCRIPTION
**Pull request by PAKovalev**

### CODE REVIEW

File: packages/map3d/src/components/Map3dCanvas/Map3dCanvas.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import {Map3dPropsInterface} from '../../interfaces';
+import {Map3dRender} from '../Map3dRender';
+
+import * as styled from './Map3dCanvas.styled';
+
+interface PropsInterface extends Map3dPropsInterface {}
+
+const Map3dCanvas: FC<PropsInterface> = (props) => {
+  const [isCanvasReady, setIsCanvasReady] = useState<boolean>(false);
```
Comment (Daboo1998) [inline packages/map3d/src/components/Map3dCanvas/Map3dCanvas.tsx]: No need for genric if you set default value

File: packages/map3d/src/components/Map3dCanvas/Map3dCanvas.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  const mapRef = useRef<HTMLCanvasElement | null>(null);
```
Comment (Daboo1998) [inline packages/map3d/src/components/Map3dCanvas/Map3dCanvas.tsx]:
```suggestion
  const mapRef = useRef<HTMLCanvasElement>(null);
```
would also work

File: packages/map3d/src/interfaces/planet.interface.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export interface PlanetInterface {
+  uuid: string;
+  owner: string;
+  name: string;
+  image?: string | undefined;
```
Comment (Daboo1998) [inline packages/map3d/src/interfaces/planet.interface.ts:5]:
```suggestion
  image?: string;
```

why repeat yourself? :)

File: packages/map3d/src/interfaces/props.interface.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import {PlanetInterface} from './planet.interface';
+
+export interface Map3dPropsInterface {
+  currentUserId: string | undefined;
```
Comment (Daboo1998) [inline packages/map3d/src/interfaces/props.interface.ts]: Are you sure not: 
```suggestion
  currentUserId?: string;
```
?

File: packages/map3d/src/interfaces/props.interface.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+  items: PlanetInterface[];
+  connections: Record<string, {id: string}[]>;
+  getImageUrl(imageUrlOrHash: string | undefined | null): string | null;
```
Comment (Daboo1998) [inline packages/map3d/src/interfaces/props.interface.ts:7]:
```suggestion
  getImageUrl(imageUrlOrHash?: string | null): string | null;
```
Not better?

**Review by Daboo1998 — APPROVED** [review]

**Review by aminbksi — APPROVED** [review]

File: packages/map3d/src/interfaces/props.interface.ts (hunk 5 above)
Comment (PAKovalev) [reply packages/map3d/src/interfaces/props.interface.ts:7]: `imageUrlOrHash?` means that prop in not required.

File: packages/map3d/src/interfaces/planet.interface.ts (hunk 3 above)
Comment (PAKovalev) [reply packages/map3d/src/interfaces/planet.interface.ts:5]: It is two different cases. There is no field at all and prop is undefined.

Comment (Daboo1998) [reply packages/map3d/src/interfaces/planet.interface.ts:5]: but it doesn't stop you from setting it as undefined explicitly 😊

```
image?: string;
```
is different from 
``` 
image: string | undefined
```
but not from 
```
image?: string | undefined;
```

File: packages/map3d/src/interfaces/props.interface.ts (hunk 5 above)
Comment (Daboo1998) [reply packages/map3d/src/interfaces/props.interface.ts:7]: yes I know, That's why I am just asking if its not better 😉

unless you really need the user to explicitly say that its undefined (or the parameter passed)

**Pull request merged**